File: SPOJ/NSTEPS.py
N = int(raw_input())

for i in range(N) :
    x  = int(input())
    y = int(input())
    if x % 2 == 0 and y % 2 == 0 :
	print x + y
    elif x % 2 != 0 and y % 2 != 0 :
	print x + y - 1
    else :
	print "No Number"

File: SPOJ/NSTEPS.c
#include <stdio.h>
int main()
{
    int N, x, y;
    scanf( "%d", &N );
    while( N-- ){
	scanf( "%d %d", &x, &y );
	if( (x % 2 == 0) && (y % 2 == 0) && (x >= y) && (x - y <= 2) ){
	    printf( "%d\n", (x + y) );
	}
	else if( (x % 2 != 0) && (y % 2 != 0) && (x >= y) && (x - y <= 2) ){
	    printf( "%d\n", (x + y - 1) );
	}
	else{    
	    printf( "No Number\n" );
	}
    }
    return 0;
}
